src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -341,6 +347,20 @@ public byte[] addressBytes(Inet6Address inet6Address) {
                 }
         );
         init();
+        // DNS cache is cleared before the checkpoint; application restored at a later point
+        // or in a different environment should query DNS again.
+        checkpointListener = new Resource() {
+            @Override
+            public void beforeCheckpoint(Context<? extends Resource> context) throws Exception {
+                cache.clear();
+                expirySet.clear();
+            }
+
+            @Override
+            public void afterRestore(Context<? extends Resource> context) throws Exception {
+            }
+        };
+        Core.getGlobalContext().register(checkpointListener);
```

Comment:
For JDK internal Resources, please prefer jdk.internal.crac.Core.getJDKContext() https://github.com/openjdk/crac/blob/crac/src/java.base/share/classes/jdk/internal/crac/Core.java#L41